hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java

```diff
@@ -344,13 +367,37 @@ public void registerConfigLocally(IdentifiedDataSerializable newConfig, ConfigCh
             if (currentConfig == null) {
                 nodeEngine.getWanReplicationService().addWanReplicationConfig(config);
             }
+        } else if (newConfig instanceof NamespaceConfig) {
+            NamespaceConfig config = (NamespaceConfig) newConfig;
+            // Deliberately overwrite existing
+            currentConfig = namespaceConfigs.put(config.getName(), config);
+            nodeEngine.getNamespaceService().addNamespaceConfig(config);
         } else {
             throw new UnsupportedOperationException("Unsupported config type: " + newConfig);
         }
         checkCurrentConfigNullOrEqual(configCheckMode, currentConfig, newConfig);
         persist(newConfig);
     }
 
+    /**
+     * Register a dynamic configuration in a local member.
+     *
+     * @param newConfig       Configuration to register.
+     * @throws UnsupportedOperationException when given configuration type is not supported
+     */
+    @SuppressWarnings("checkstyle:methodlength")
+    public void deregisterConfigLocally(IdentifiedDataSerializable newConfig) {
+        if (newConfig instanceof NamespaceConfig) {
+            NamespaceConfig config = (NamespaceConfig) newConfig;
+            if (namespaceConfigs.remove(config.getName(), newConfig)) {
+                nodeEngine.getNamespaceService().removeNamespaceConfig(config);
```

Comment:
The call to `namespaceService.removeNamespaceConfig` should be always executed, since we are supposed to allow for removal of statically configured namespaces. The current code with the condition above will not work as expected in this scenario:

- start member with statically configured namespace `ns1`
- client invokes `hz.getConfig().getNamespacesConfig().removeNamespace("ns1")`
- since there is no entry in dynamically configured `namespaceConfigs` map maintained in `ClusterWideConfigurationService`, namespace is not removed from `NamespaceService`

Reply:
Addressed in https://github.com/vbekiaris/hazelcast/commit/cafc0048ba946af66a094d8832bc1dcfb59eed5d